build/flash_emmc.sh

```diff
@@ -0,0 +1,185 @@
+[[ -f /usr/lib/u-boot/platform_install.sh ]] && source /usr/lib/u-boot/platform_install.sh
+
+IMAGE=$1
+
+# script configuration
+CWD="/usr/lib/nand-sata-install"
+EX_LIST="${CWD}/exclude.txt"
+
+# read in board info
+[[ -f /etc/armbian-release ]] && source /etc/armbian-release
+
+#recognize_root
+root_uuid=$(sed -e 's/^.*root=//' -e 's/ .*$//' < /proc/cmdline)
+root_partition=$(blkid | tr -d '":' | grep ${root_uuid} | awk '{print $1}')
+root_partition_device="${root_partition::-2}"
+
+# find targets: NAND, EMMC, SATA, SPI
+emmccheck=$(ls -d -1 /dev/mmcblk* | grep -w 'mmcblk[0-9]' | grep -v "$root_partition_device");
+
+# Create boot and root file system $1 = boot, $2 = root (Example: create_armbian "/dev/nand1" "/dev/sda3")
+create_armbian()
+{
+	# create mount points, mount and clean
+	TempDir=$(mktemp -d /mnt/${0##*/}.XXXXXX || exit 1)
+	sync &&	mkdir -p ${TempDir}/bootfs ${TempDir}/rootfs
+	[[ -n $2 ]] && ( mount -o compress-force=zlib $2 ${TempDir}/rootfs || mount $2 ${TempDir}/rootfs )
+	[[ -n $1 && $1 != "spi" ]] && mount $1 ${TempDir}/bootfs
+	rm -rf ${TempDir}/bootfs/* ${TempDir}/rootfs/*
+
+	# sata root part
+	# UUID=xxx...
+	satauuid=$(blkid -o export $2 | grep -w UUID)
+
+	# SD card boot part -- wrong since more than one entry on various platforms
+	# UUID=xxx...
+	sduuid=$(blkid -o export /dev/mmcblk*p1 | grep -w UUID | grep -v "$root_partition_device")
+
+	# calculate usage and see if it fits on destination
+	USAGE=$(df -BM | grep ^/dev | head -1 | awk '{print $3}' | tr -cd '[0-9]. \n')
+	DEST=$(df -BM | grep ^/dev | grep ${TempDir}/rootfs | awk '{print $4}' | tr -cd '[0-9]. \n')
+	if [[ $USAGE -gt $DEST ]]; then
+        echo "Partition too small. Needed: $USAGE MB Avaliable: $DEST MB"
+		umountdevice "$1"; umountdevice "$2"
+		exit 1
+	fi
+
+	# creating rootfs
+	#rsync -avrltD --delete --exclude-from=$EX_LIST / ${TempDir}/rootfs
+	echo "Mounting image ..."
+	mkdir -p /mnt/image_for_emmc/
+	umount /mnt/image_for_emmc/ 2>/dev/null || true
+	mount -o loop,offset=4194304 "$IMAGE" /mnt/image_for_emmc
+	echo "Running rsync to eMMc ..."
+	rsync -arltD --delete --no-inc-recursive --info=progress2 --exclude-from=$EX_LIST /mnt/image_for_emmc/ ${TempDir}/rootfs
+	umount /mnt/image_for_emmc/
+
+	# creating fstab from scratch
+	echo "Rewriting fstab ..."
+	rm -f ${TempDir}/rootfs/etc/fstab
+	mkdir -p ${TempDir}/rootfs/etc ${TempDir}/rootfs/media/mmcboot ${TempDir}/rootfs/media/mmcroot
+
+	# Restore TMP and swap
+	echo "# <file system>					<mount point>	<type>	<options>							<dump>	<pass>" > ${TempDir}/rootfs/etc/fstab
+	echo "tmpfs						/tmp		tmpfs	defaults,nosuid							0	0" >> ${TempDir}/rootfs/etc/fstab
+	grep swap /etc/fstab >> ${TempDir}/rootfs/etc/fstab
+
+	# Boot from eMMC, root = eMMC or SATA / USB
+	local targetuuid=$emmcuuid
+	local choosen_fs=$eMMCFilesystemChoosen
+	echo "Finishing full install to eMMC."
+
+	# fix that we can have one exlude file
+	cp -R /boot ${TempDir}/bootfs
```

Comment:
Currently, this means that the image flashed must be the ~same as the image used for flashing - or at least they must be using the same kernel ...